composer/callbacks/checkpoint_saver.py

```diff
@@ -319,16 +333,19 @@ def _save_checkpoint(self, state: State, logger: Logger):
 
         # save the checkpoint to the filename
         filename_with_placeholders = self.filename.format(state, is_deepspeed, keep_placeholders=True)
+        save_filename = checkpoint.get_save_filename(state, filename_with_placeholders)
```

Comment:
do we need refactor? It looks like `save_checkpoint` returns name? We can change that behavior to return on all ranks if desired?

Reply:
Yep - we need to know the filename and add it to state before the local checkpoint is saved - otherwise the checkpoint will be missing reference to itself when resumed

Reply:
Added comment in 5770ee40a898dcc5bd571e9b4143da8407df5f51 since this didn't feel obvious